dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/outlier/data/OutlierQueryParser.java

```diff
@@ -164,4 +168,54 @@ private List<Period> getPeriods(RelativePeriodEnum relativePeriod, Date relative
         .map(pe -> (Period) pe)
         .toList();
   }
+
+  /**
+   * The method retrieves the enumerator item related to the column for order by statement.
+   *
+   * @param orderBy is the order by field. It has to be the same as header item name.
+   * @return the {@link Order}.
+   */
+  private Order getOrderBy(String orderBy) {
```

Comment:
We could, probably, replace all these "IFs" by creating a method inside the enum Order that returns the respective Order based on the `orderBy` (which should be renamed to `headerName`).

Reply:
done